public/index.js

```diff
@@ -26,11 +26,34 @@ var clearContents = function () {
 }
 
 /**
- * Generates the random gift page building all necessary elements
+ * Generate the random gift page building all necessary elements
  */
 var generateGiftPage = function () {
   var hello = document.createElement('p');
   var helloText = document.createTextNode('I am the gift page');
   hello.appendChild(helloText);
   main.appendChild(hello);
+  var giftButton = document.createElement('BUTTON');
```

Comment:
naming: we need to distinguish this gift button from other gift buttons that are planned